dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -1059,42 +1158,9 @@ def _update_histogram(self, df_series: pd.Series) -> None:
         self._stored_histogram["current_loss"] = histogram_loss
         self._stored_histogram["total_loss"] += histogram_loss
 
-    def _histogram_for_profile(
-        self, histogram_method: str
+    def _regenerate_histogram(
```

Comment:
broke out `_histogram_for_profile()` function into two functions in the code so that we could call `_regenerate_histogram` on either `self` or `other_profile`.

The idea here is that to calculate PSI the edges in the historgram need to be the same between `self` and `other_profile`. In order to accomplish, prior to calling `_calculate_psi` potentially one or both `obj._stored_histograms` need to be updated. 

This is the purpose of the refactor and breaking out into `_regenerate_histogram`.